### PR TITLE
Accept only files for qvm-open-in-(d)vm

### PR DESCRIPTION
**Pull request by alimirjamali**
fixes: https://github.com/QubesOS/qubes-issues/issues/9360

### CODE REVIEW

**Comment by alimirjamali** [comment]
Accepting URLs should be added to the conditions. Fix will come in few minutes...

**Comment by alimirjamali** [comment]
According to tests:
```
In qubes-rpc/qvm-open-in-vm line 73:
elif [[ ! "$filename" =~ (https?|ftp|mailto|irc|rsync)://.*$ ]] && \
     ^-- SC3010 (warning): In POSIX sh, [[ ]] is undefined.
For more information:
  https://www.shellcheck.net/wiki/SC3010 -- In POSIX sh, [[ ]] is undefined.
```
Instead of upgrading the script to bash to allow regex, I believe it would be better to revert to the conventional `expr` URL evaluation which is already used in the script. Handling malformed or invalid URLs was not the main subject of this PR from the beginning.

File: qubes-rpc/qvm-open-in-vm [hunk 1]
```diff
@@ -64,6 +64,21 @@ if [ -z "$target" ] || [ -z "$filename" ]; then
     usage
 fi
 
+if [ -d "$filename" ]; then
+    printf 'Error! "%s" is a directory.\n' "$filename"
+    usage
+elif [ -L "$filename" ]; then
```
Comment (marmarek) [inline qubes-rpc/qvm-open-in-vm:70]: Why reject symlinks? They open just fine (but when saving modified file, symlink is replaced with a copy of the file, and the actual target file remains unchanged).

Comment (alimirjamali) [reply qubes-rpc/qvm-open-in-vm]: > Why reject symlinks? They open just fine (but when saving modified file, symlink is replaced with a copy of the file, and the actual target file remains unchanged).

I was not aware of that. Symlinks are now allowed in the amended PR.

**Review by marmarek — APPROVED** [review]

**Pull request merged**